### PR TITLE
Part1 tasks

### PR DESCRIPTION
**Pull request by hrytsiuk**

### CODE REVIEW

File: part1/color.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'pry'
+
+class Color
+  attr_reader :r, :g, :b
```
Comment (Max43) [inline part1/color.rb:8]: is you initializer(constructor) receive more than two params use optional parameter 

```
def initialize(params = {})
....
```

File: part1/color.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  def initialize(r, g, b)
+    @r = r
+    @g = g
+    @b = b
+  end
+
+  def brightness_index
+    (299*r + 587*g + 114*b) / 1000
```
Comment (Max43) [inline part1/color.rb]: always try to avoid magic number in you codebase. Use constants instead:
`SOME_NUMBER = 299`

File: part1/color.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def brightness_difference(another)
+    (brightness_index - another.brightness_index).abs
+  end
+
+  def hue_difference(another)
+    (r - another.r).abs + (g - another.g).abs + (b - another.b).abs
+  end
+
+  def enough_contrast?(another)
+    brightness_difference(another) > 125 && hue_difference(another) > 500
```
Comment (Max43) [inline part1/color.rb]: same here(constants)

File: part1/color.rb (hunk 1 above)
Comment (hrytsiuk) [reply part1/color.rb:8]: can I use options = {} instead of params = {}?

File: part1/restaurant.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Restaurant
```
Comment (Max43) [inline part1/restaurant.rb:1]: forgot about attribute accessor method , for @menu

File: part1/restaurant.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  def initialize(menu)
+    @menu = menu
+  end
+
+  def cost(*orders)
```
Comment (Max43) [inline part1/restaurant.rb:10]: Good, you know how to use it

but pls take a look https://medium.freecodecamp.org/rubys-splat-and-double-splat-operators-ceb753329a78

File: part1/restaurant.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    cost = orders.map do |order|
+      order[:rice] * @menu[:rice] + order[:noodles] * @menu[:noodles]
```
Comment (Max43) [inline part1/restaurant.rb]: it looks like you broke Law of Demeter with this, as you use only `menu[:rice]`  `@menu[:noodles]` but have object menu, which kind of excessive

pls look read about  Law of Demeter http://www.virtuouscode.com/2011/07/05/demeter-its-not-just-a-good-idea-its-the-law/

File: spec/array_of_integer.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
+require_relative '../part1/array_of_integer'
+require 'pry'
+
+RSpec.describe 'array_of_integer?' do
+  subject { [] }
```
Comment (Max43) [inline spec/array_of_integer.rb]: just for you to feel the difference between `subject` and `let()`
https://stackoverflow.com/questions/38437162/whats-the-difference-between-rspecs-subject-and-let-when-should-they-be-used

and maybe you will decide that you need to use let here)) ;)

File: spec/array_of_integer.rb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  context 'all items are integer' do
+    before { subject.push(1,2,4) }
+    it 'returns true' do
+      expect(array_of_integer?(subject)).to eq true
+    end
+  end
+
+  context 'items are of different type' do
+    before { subject.push(:a, "c", 1) }
```
Comment (Max43) [inline spec/array_of_integer.rb]: bettter not to use subject fot such cases

and you simple let

let(array) {[:a, "c", 1]}

File: spec/color_spec.rb [hunk 9]
```diff
@@ -0,0 +1,54 @@
+require_relative '../part1/color'
+require 'pry'
+
+RSpec.describe Color do
+
+  let(:rgb_color) { described_class.new(13, 21, 34) }
+  let(:another_color) { described_class.new(55, 89, 144) }
+  index = 20
```
Comment (Max43) [inline spec/color_spec.rb]: use let, try to avoid such assignment in rspec

Comment (Max43) [reply spec/color_spec.rb]: if it not working for your case

try let! 

https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let let us let!

File: spec/color_spec.rb [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    context 'another color is exists' do
+      it 'returns value of difference' do
+        expect(rgb_color.hue_difference(another_color)).to eq(220)
+      end
+    end
+  end
+
+  describe '#enough_contrast?' do
+    context 'is enough contrast' do
+    let(:rgb_color) { described_class.new(42, 42, 42) }
```
Comment (Max43) [inline spec/color_spec.rb]: alignment

File: spec/length_finder_spec.rb [hunk 11]
```diff
@@ -0,0 +1,26 @@
+require_relative '../part1/length_finder'
+require 'rspec'
+
+RSpec.describe '.length_finder' do
+  context 'array contains words' do
+  let(:array) { ['Ruby', 'on', 'Rails'] }
```
Comment (Max43) [inline spec/length_finder_spec.rb]: alignment

let must be inside context

File: spec/number_shuffle_spec.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
+require 'rspec'
+require_relative '../part1/number_shuffle'
+
+RSpec.describe 'number_shuffle' do
+  let(:number) { 123 }
+  array = [123, 132, 213, 231, 312, 321]
```
Comment (Max43) [inline spec/number_shuffle_spec.rb]: avoid use let

**Pull request merged**